src/components/ExecuteButton.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect } from 'react';
+
+import PropTypes from 'prop-types';
+import { downloadPlaybook } from '../api';
+import {
+    Button, Modal, TextContent, Text, TextVariants } from '@patternfly/react-core';
+import { TableHeader, Table, TableBody, TableVariant } from '@patternfly/react-table';
+import { CheckCircleIcon } from '@patternfly/react-icons';
+import { Skeleton } from '@redhat-cloud-services/frontend-components';
+import './ExecuteButton.scss';
+
+const ExecuteButton = ({ isLoading, data, getConnectionStatus, remediationId, issueCount }) => {
+    const [ open, setOpen ] = useState(false);
+    const [ isUserEntitled, setIsUserEntitled ] = useState(false);
+    useEffect(() => {
+        window.insights.chrome.auth.getUser().then(user => setIsUserEntitled(user.entitlements.smart_management.is_entitled));
+    }, []);
+
+    const styledConnectionStatus = (status) => (
+        status === 'connected'
+            ? (<div>
+                <CheckCircleIcon
+                    className="ins-c-remediations-reboot-check-circle ins-c-remediations-connection-status"
+                    aria-label="connection status"/>
+                Ready
+            </div>)
+            : status
```

Comment:
status is an enum that is not intended to be displayed directly to users. We'll need to somehow "translate" the enum value to a human readable string. The full list of enum values can be found here: https://github.com/RedHatInsights/insights-remediations/blob/master/src/api/openapi.yaml#L925

Some of these human readable labels are listed here: https://marvelapp.com/j3b9b7e/screen/63670941

Here's my suggestion:
* disconnected - Connection issue. Receptor not connected (see mocks)
* no_executor - Cannot remediate - Direct connection. Connect your systems to Satellite to automatically remediate. (see mocks)
* no_source - Cannot remediate - Satellite not configured. Satellite not registered for Playbook execution.
* no_receptor - Cannot remediate - Receptor not configured. Configure Receptor to automatically remediate

Reply:
Good point, I would make this into Object mapper.